lib/Runtime/Library/ConcatString.inl

```diff
@@ -82,6 +91,8 @@ namespace Js
         // Allow slots to be garbage collected if no more refs.
         ClearArray(m_slots, N);
 
+        LiteralStringWithPropertyStringPtr::ConvertString(this);
```

Comment:
We could, but I didn't like the idea of doing work with fields of the old class after changing type.